plugins/basic/python/tk_maya_basic/plugin_engine.py

```diff
@@ -53,25 +52,29 @@ def bootstrap(sg_user):
     toolkit_mgr.base_configuration          = manifest.base_configuration
     toolkit_mgr.bundle_cache_fallback_paths = [bundle_cache_path]
 
-    logger.info("Starting the %s engine." % manifest.engine_name)
-
     # Remove the custom logging handler now that the engine will take over logging.
     sgtk.LogManager().root_logger.removeHandler(plugin_logging_handler)
 
-    # Ladies and Gentlemen, start your engines!
+    # Bootstrap a toolkit instance asynchronously in a background thread,
+    # followed by launching the engine synchronously in the main application thread.
+    toolkit_mgr.async_bootstrap = True
+
     # Before bootstrapping the engine for the first time around,
     # the toolkit manager may swap the toolkit core to its latest version.
-    toolkit_mgr.bootstrap_engine(manifest.engine_name)
-
-    # Re-import the toolkit core to ensure usage of a swapped in version.
-    import sgtk
+    toolkit_mgr.bootstrap_engine(manifest.engine_name,
+                                 progress_callback=progress_callback,
+                                 completed_callback=completed_callback,
+                                 failed_callback=failed_callback)
 
 
 def shutdown():
     """
     Shuts down the running engine.
     """
 
+    # Re-import the toolkit core to ensure usage of a swapped in version.
+    import sgtk
+
     logger = sgtk.LogManager.get_logger(PLUGIN_PACKAGE_NAME)
```

Comment:
Using `__file__` would display information/warning/error messages like this one to the user in Maya script editor:

> // Shotgun: plugin_engine: Stopping the tk-maya engine. //

Using `PLUGIN_PACKAGE_NAME` replaces `plugin_engine` by the more descriptive plug-in package name, for the time being `tk_maya_basic`.

Ideally, tk-core engine logging and tk-maya should allow setting the message prefix rather than forcing the not-very-user-friendly predefined format `Shotgun: %(basename)s: `. This would allow using as prefix the plug-in name defined in the `info.yml` file.